### DISCUSSION TITLE
A better method to force repaint?

### DISCUSSION THREAD

**Discussion by astrojhgu** [emilk/egui#1948] [Q&A] [answered]
I need the GUI to repaint when a set of data that is to be plotted in the window is updated in a background threshold.

Currently I set a `ctx: Arc<Mutex<Option<egui::Context>>>` member in my ` struct MyApp`, the value of which will be set when the `fn update` is first called.

When the thread that is responsible for preparing the data prepares the data, it will put the data into a buffer inside `MyApp` and call the `ctx.lock().request_repaint()` manually.

Is there a better way to do this?

**Comment by emilk** [comment, marked as answer]
You could send the data over a channel instead, or use something like [`poll_promise`](https://docs.rs/poll-promise).

**Reply by cgo** [reply]
I'm having a similar question, I think; I am running a separate thread that is doing some rendering to memory; then when that is done, I call a callback from that thread, which copies the data into a texture and then calls request_repaint() on a /clone/ of the context. Is this supposed to work also, and is it expected to trigger an immediate redraw? My window content seems to be only updated when I move the mouse on the window. It does not automatically update.
I suppose that I am holding it wrong, any comment on that, any ideas?

**Reply by cgo** [reply]
I think I just found the answer myself -- reading some other posts about request_repaint(), I realized I had to clone the Frame, not the Context. The Frame worked -- sorry for the noise!

**Reply by emilk** [reply]
Since egui 0.18 using `Context::request_repaint` is enough (and `Frame::request_repaint` is gone).
